### PR TITLE
Fallback to global variable

### PR DESCRIPTION
**Pull request by roberttod**
- Fallback to using window.__queueThat__ when localStorage throws errors or doesn't persist
- Add some functional tests
- Use phantomjs as default test runner (much less of the browser did full reload bug)

Tested the functional test in incognito safari - seems to work fine now.

### CODE REVIEW

**Comment by roberttod** [comment]
@davidpadbury (addresses https://github.com/QubitProducts/queue-that/issues/12)

**Comment by davidpadbury** [comment]
Why install a global? For multiple instances of queue-that on a page?

Phantom's probably fine, but out of interest - what's this full reload bug?

**Comment by roberttod** [comment]
Yes, the global is for multiple instances and perhaps for inspection... but actually maybe we don't need it?

I actually realised that this would break multiple queues right now because of how that is done, going to add a functional test to cover that case.

**Comment by roberttod** [comment]
Real browsers seem to constantly
- say that `one of your tests did a full page reload`
- detect that you are not interacting with the browser, often breaking async tests or making test running really slow

**Comment by roberttod** [comment]
I use a global variable to recreate the functionality of localStorage (because that uses a global store). i.e. if there are two instances with the same label then they can push to the same queue (still not sure if there is a use case but if someone uses it like that then I think this would be the desired behaviour).

File: lib/global-variable-adapter.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+var createLocalStorageAdapter = require('./local-storage-adapter')
+
+module.exports = createGlobalVariableAdapter
+
+function createGlobalVariableAdapter (queueName) {
+  window.__queueThat__ = window.__queueThat__ || {}
+
+  var localStorageAdapter = createLocalStorageAdapter(queueName)
```
Comment (davidpadbury) [inline lib/global-variable-adapter.js]: I guess I find this a little weird, but I get what is going on. I think I initially expected there would be entirely separate `localStorage` vs `variable` adapters, however this is just patching the local storage adapter to use a variable so we can reuse it's other functionality.

**Comment by davidpadbury** [comment]
:+1:

File: lib/global-variable-adapter.js (hunk 1 above)
Comment (roberttod) [inline lib/global-variable-adapter.js]: Yeh, I think the code could be a lot neater. I am thinking that this idea of a pluggable adapter might be a bit contrived and maybe there should just be one storage module that will fall back to global storage automatically. I will probably rewrite in the future.

**Pull request merged**